protocol-jackson/src/test/java/io/camunda/zeebe/protocol/jackson/record/IntentTypeIdResolverTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.protocol.jackson.record;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.BeanDeserializerFactory;
+import com.fasterxml.jackson.databind.deser.DefaultDeserializationContext;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.Intent;
+import java.io.IOException;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.EnumSource.Mode;
+
+final class IntentTypeIdResolverTest {
+
+  /**
+   * This test checks that every known intent type is handled. It doesn't validate the correctness
+   * of the result - its goal is to be a smoke test to make sure no intents are forgotten
+   *
+   * @deprecated to be removed when intent classes are directly mapped via the {@link ValueType}
+   *     enum
+   */
+  @EnumSource(
+      value = ValueType.class,
+      names = {"NULL_VAL", "SBE_UNKNOWN"},
+      mode = Mode.EXCLUDE)
+  @ParameterizedTest
+  void shouldHandleEveryKnownValueType(final ValueType type) throws IOException {
+    // given
+    final ObjectMapper mapper = new ObjectMapper();
+    final DefaultDeserializationContext.Impl baseContext =
+        new DefaultDeserializationContext.Impl(BeanDeserializerFactory.instance);
+    final DefaultDeserializationContext context =
+        baseContext.createInstance(
+            mapper.getDeserializationConfig(),
+            mapper.createParser("{}"),
+            mapper.getInjectableValues());
+    final IntentTypeIdResolver resolver = new IntentTypeIdResolver();
+    final JavaType resolvedType = resolver.typeFromId(context, resolver.idFromValue(type));
```

Comment:
Why only some of them to var and not all? :upside_down_face:

Reply:
`var` was not intended to replace all local variable type declarations. The original JEP came with a [style guide](https://openjdk.java.net/projects/amber/LVTIstyle.html) that proposed to use it in specific cases where the type is already clear from the rest of the line or statement, or where you otherwise don't care about the type. For example, calling a constructor, or a method who's name clearly shows what type it returns. In some cases, understandability would be reduced by inferring the type.

I think the `DefaultDeserializationContext` and `JavaType` could offer some additional insight into what this code does, but I don't care too much for it. Feel free to change it how you think it reads best.